Dockerfile

```diff
@@ -1,82 +1,90 @@
-# PYTHON image
-# Use the official Docker Python image because it has the absolute latest bugfix version of Python
-# it has the absolute latest system packages
-# it’s based on Debian Bookworm (Debian 12), released June 2023
-# Initial Image size is 51MB
-# At the end Image size is 156MB
+################################################################################
+# BASE                                                                         #
+################################################################################
+FROM python:3.11-slim-bookworm as base
+
+ARG POETRY_VERSION=1.8.3
+ARG UID=1000
+ARG GID=1000
 
-# I did not recommed using an alpine image because it lacks the package installer pip and the support for installing
-# wheel packages, which are both needed for installing applications like Pandas and Numpy.
+# Where python should look for packages and modules when using import
+ENV PYTHONPATH="/app"
 
-# The base layer will contain the dependencies shared by the other layers
-FROM python:3.11-slim-bookworm as base
+# Ensure the stdout and stderr streams are sent straight to terminal
+ENV PYTHONUNBUFFERED=1 
+
+# Extend the socket timeout. Default would be 15s
+ENV  PIP_DEFAULT_TIMEOUT=100 
 
-# Allowing the argumenets to be read into the dockerfile. Ex:  .env > compose.yml > Dockerfile
-ARG POETRY_VERSION
-# true = development / false = production
-ARG DEV
+RUN groupadd -g ${GID} -o app
+RUN useradd -m -d /app -u ${UID} -g ${GID} -o -s /bin/bash app
+
+# RUN apt-get update -yqq && apt-get install -yqq --no-install-recommends \
+#   vim-tiny
```

Comment:
Should this be removed? Or do we want to add a comment explaining this is how to add packages if you need them?